osu.Game/Online/API/Requests/SearchBeatmapSetsRequest.cs

```diff
@@ -89,27 +68,10 @@ protected override WebRequest CreateWebRequest()
 
             req.AddParameter("sort", $"{SortCriteria.ToString().ToLowerInvariant()}_{directionString}");
 
-            if (cursor != null)
+            cursor?.Properties.ForEach(x =>
             {
-                if (cursor.Title != null)
-                    req.AddParameter("cursor[title.raw]", cursor.Title);
-                if (cursor.Artist != null)
-                    req.AddParameter("cursor[artist.raw]", cursor.Artist);
-                if (cursor.Difficulty != null)
-                    req.AddParameter("cursor[beatmaps.difficultyrating]", cursor.Difficulty);
-                if (cursor.Ranked != null)
-                    req.AddParameter("cursor[approved_date]", cursor.Ranked);
-                if (cursor.Rating != null)
-                    req.AddParameter("cursor[rating]", cursor.Rating);
-                if (cursor.Plays != null)
-                    req.AddParameter("cursor[play_count]", cursor.Plays);
-                if (cursor.Favourites != null)
-                    req.AddParameter("cursor[favourite_count]", cursor.Favourites);
-                if (cursor.Relevance != null)
-                    req.AddParameter("cursor[_score]", cursor.Relevance);
-
-                req.AddParameter("cursor[_id]", cursor.Id);
-            }
+                req.AddParameter("cursor[" + x.Key + "]", x.Value?.ToString());
+            });
```

Comment:
Definitely better in my opinion, but I'd make this an extension method of `WebRequest` so that it can be reused in other places:

```csharp
public static void AddCursor(this WebRequest webRequest, Cursor cursor)
{
    // this impl here
}
```

Reply:
Not sure whether I understand correctly, so double checking
I'm guessing you didn't mean as in adding extension method directly in osu-framework's `osu.Framework.IO.Network.WebRequest` class
Should it be still part of `SearchBeatmapSetsRequest.cs` as in something like this
```
public static class ExtensionClass
    {
        public static void AddCursor(this WebRequest webRequest, SearchBeatmapSetsRequest.Cursor cursor)
        {
            cursor?.Properties.ForEach(x =>
            {
                webRequest.AddParameter("cursor[" + x.Key + "]", x.Value.ToString());
            });
        }
    }
```
Or is there some namespace inside osu.Game for storing extensions like this already
Not sure where I'm expected to add it and whether to move Cursor class aswell

Reply:
Or just do something like creating `WebRequestExtensions.cs` in `osu.Game.Extensions` namespace

Reply:
The latter. The cursor is osu!-specific and adding anything to do with it to framework does not make sense.

Reply:
I'm not sure whether this is a clean solution, at least structure-wise
But committed for now